sdv/metadata.py

```diff
@@ -116,6 +116,76 @@ def _analyze_relationships(self):
                         self._child_map[parent].add(table)
                         self._parent_map[table].add(parent)
 
+    def _validate_dataset_structure(self):
+        """Make sure that all the tables have at most one parent."""
+        for table in self.get_tables():
+            if len(self.get_parents(table)) > 1:
+                raise ValueError('Some tables have multiple parents, which is not supported yet.')
+
+    def _validate_table(self, table_name, table_meta, tables=None):
+        """Validate metadata table.
+
+        Args:
+            table_name (str):
+                Table name to be validated.
+            table_meta (dict):
+                Table metadata to be validated.
+            tables (dict):
+                Dictionary with the name of the table and their data.
+                If it's provided, assert metadata types are valid and
+                metadata fields exist in data columns.
+        """
+        dtypes = self.get_dtypes(table_name, ids=True)
+
+        # Primary key field exists and its type is 'id'
+        primary_key = table_meta.get('primary_key')
+        if primary_key:
+            pk_field = table_meta['fields'].get(primary_key)
+
+            if not pk_field:
+                raise MetadataError('Primary key is not an existing field.')
+
+            if pk_field['type'] != 'id':
+                raise MetadataError('Primary key is not of type `id`.')
+
+        if tables:
```

Comment:
`if table_data is not None`

Also, let's be more verbose about the failing fields.
Try to capture a `ValueError` and, if so, raise a MetadataError indicating that field xxx has a a wrong type.
Otherwise the user will see something line `ValueError: invalid literal for int() with base 10: 'a'` and will have not clue about which column this comes from.

Do this by concatenating the ValueError message to the user-friendly message, and raising from None to avoid the extra long traceback:

```
except ValueError as ve:
    message = 'Invalid values found in column {} of table {}: {}'.format(
        column, table_name, ve)
    raise MetadataError(message) from None
```